lib/src/rewrite.rs

```diff
@@ -796,6 +782,190 @@ pub fn move_commits(
     })
 }
 
+#[derive(Default)]
+pub struct DuplicateCommitsResult {
```

Comment:
nit: `Result` -> `Stats` ?

It's not actually "stats", but seems better than calling non-`Result` type as `Result`.